lib/ddtrace/contrib/grpc/patcher.rb

```diff
@@ -0,0 +1,62 @@
+# requirements should be kept minimal as Patcher is a shared requirement.
+
+module Datadog
+  module Contrib
+    module GRPC
+      SERVICE = 'grpc'.freeze
+
+      # Patcher enables patching of 'grpc' module.
+      module Patcher
+        include Base
+        register_as :grpc, auto_patch: true
+        option :tracer, default: Datadog.tracer
+        option :service_name, default: SERVICE
+
+        @patched = false
+
+        module_function
+
+        def patch
+          return false unless compatible?
+          return @patched if @patched
+
+          require 'ddtrace/ext/grpc'
+          require 'ddtrace/propagation/grpc_propagator'
+          require 'ddtrace/contrib/grpc/datadog_interceptor'
+          require 'ddtrace/contrib/grpc/interception_context'
+
+          add_pin
+          prepend_interceptor
+
+          @patched = true
+        rescue StandardError => e
+          Datadog::Tracer.log.error("Unable to apply gRPC integration: #{e}")
+        ensure
+          @patched
+        end
+
+        def compatible?
+          defined?(::GRPC::VERSION) && Gem::Version.new(::GRPC::VERSION) >= Gem::Version.new('0.10.0')
+        end
+
+        def patched?
+          @patched
+        end
+
+        def add_pin
+          Pin.new(
+            get_option(:service_name),
+            app: 'grpc',
+            app_type: 'grpc',
+            tracer: get_option(:tracer)
+          ).onto(::GRPC)
+        end
+
+        def prepend_interceptor
+          ::GRPC::InterceptionContext
+            .prepend(::GRPC::InterceptionContext::InterceptWithDatadog)
```

Comment:
What versions of Ruby are we supporting with GRPC? A limited set is okay, but I can see this won't work with 1.9.3, and might not work with 2.0 (I think you might have to do `send(:prepend)`?)

Reply:
Due to it's use of keyword arguments, [only 2.x is supported](https://github.com/grpc/grpc/tree/master/src/ruby#prerequisites).

I think if you're uncomfortable with [modifying the existing object](https://github.com/DataDog/dd-trace-rb/pull/403#discussion_r183079568) then we could simply require the more involved configuration. Otherwise, I'll look into and test some earlier `2.x` rubies.

Reply:
Requiring the more involved configuration seems "cleaner" and simpler, however, I think for now, prepending into the GRPC module is okay, for the sake of having some out of the box, easy GRPC tracing.

Let's just verify its working with 2.0+ for now.

Reply:
That seems fine with me; seems like a feature that should require sufficiently new Ruby anyways. We'll just have to be sure to add that to the documentation.